core/trino-main/src/test/java/io/trino/sql/analyzer/TestTypeSignatureTranslator.java

```diff
@@ -33,7 +32,7 @@ private void assertRoundTrip(String expression)
     {
         assertThat(type(expression))
                 .ignoringLocation()
-                .withComparatorForType(Comparator.comparing(identifier -> identifier.getValue().toLowerCase(Locale.ENGLISH)), Identifier.class)
+                .withComparatorForType(Comparator.comparing(Identifier::getCanonicalValue), Identifier.class)
```

Comment:
This changes semantics. Is it intentional? The canonical value per SQL standard rules uses upper case. For historical reasons, and until #17 is fixed, we continue to use lower case in many places.

Reply:
Does it matter here? I assumed that for sake of testing assertion whatever normalization (upper/lower) is fine. I changed to `getCanonicalValue` mostly to have different matching logic for delimited (uppercasing) and non-delimited (no uppercasing) identifiers. Without the change, the test passes for field names that require delimiting even without changes to `TypeSignatureTranslator`. I agree this is kinda a hack though. I can take that back and assume that regression testing is solely done via `testImplicitCastToRowWithFieldsRequiringDelimitation()`.

Reply:
Reverted